File: python/ex41.py
def iterative_sum(arr):
    result = 0
    for n in arr:
        result +=n
    
    return result

def recursive_sum(arr):
    if len(arr) == 0:
        return 0

    if len(arr) == 1:
        return arr[0]

    return arr.pop(0) + recursive_sum(arr)

print(iterative_sum([2,4,6]))
print(recursive_sum([2,4,6]))

File: typescript/Ex41.ts
export default class Ex41 {
    iterativeSum(array: number[]): number {
        let result: number = 0;

        for (let item of array) {
            result += item;
        }

        return result;
    }

    recursiveSum(array: number[]): number {
        if (array.length == 0) {
            return 0;
        }

        if (array.length == 1) {
            return array[0];
        }

        return (array.shift() || 0) + this.recursiveSum(array);
    }
}